cpp/src/arrow/util/ree_util.cc

```diff
@@ -47,6 +47,25 @@ int64_t LogicalNullCount(const ArraySpan& span) {
   return null_count;
 }
 
+template <typename RunEndCType>
+void SetLogicalNullBits(const ArraySpan& span, uint8_t* out_bitmap, int64_t out_offset) {
+  const auto& values = ValuesArray(span);
+  const auto& values_bitmap = values.buffers[0].data;
```

Comment:
buffers[0] could be `NULLPTR` here.

Reply:
`auto *values_bitmap = values.MayHaveNulls() ? buffers[0].data : NULLPTR;` is the way to go here